### PR TITLE
RS-17369: fix sankey diagram handling of non-factor data

### PR DESCRIPTION
**Pull request by chschan**
Fixes character data, so that it doesn't get ordered alphabetically. This also fixes the handling of QDate variables.

### CODE REVIEW

File: R/sankeydiagram.R [hunk 1]
```diff
@@ -83,8 +84,8 @@ SankeyDiagram <- function(data = NULL, links.and.nodes = NULL, output.data.only
         # but if table was constructed in other ways, we still want to convert text
         # to factors
         for (i in 1:ncol(data)) {
-            if (is.character(data[[i]]))
-                data[[i]] <- as.factor(data[[i]])
+            if (!is.factor(data[[i]]))
```
Comment (JustinCCYap) [inline R/sankeydiagram.R]: What case does this change address? Data other than character? Is it for numeric data?

File: .circleci/config.yml [hunk 2]
```diff
@@ -7,7 +7,7 @@ parameters:
     default: ""
   remote-deps:
     type: string
-    default: ""
+    default: cran/ggplot2@3.4.4
```
Comment (JustinCCYap) [inline .circleci/config.yml:10]: Why is this needed?

File: R/sankeydiagram.R (hunk 1 above)
Comment (chschan) [reply R/sankeydiagram.R]: Good point. This is for character and QDates. But numeric data should be excluded

File: .circleci/config.yml (hunk 2 above)
Comment (chschan) [reply .circleci/config.yml:10]: It was causing the splineforsimulatanousconfidenceintervals function to fail. ggplot2 is frozen on nightly, so this is just making it consistent. https://github.com/Displayr/r-server/blob/ea585a9518340aaa283484e131fc2ec2f41c8b3f/etc/r-packages-nightly#L41

**Review by JustinCCYap — APPROVED** [review]
LGTM
